tests/base.py

```diff
@@ -0,0 +1,196 @@
+        }
+        if original:
+            return return_value
+
+        # Reassign start date
+        return_value["start_date"] = self.START_DATE
+        return return_value
+
+    def expected_metadata(self):
+        return {
+            "campaign": {
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.FULL_TABLE
+            },
+            "content_area":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "data_extension.test emails":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "data_extension.This is a test":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "data_extension.my_test":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "data_extension.test 1":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"JoinDate"},
+            },
+            "email":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "event": {
+                self.PRIMARY_KEYS: {"SendID", "EventType", "SubscriberKey", "EventDate"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"EventDate"},
+            },
+            "folder":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "list":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "list_send":{
+                self.PRIMARY_KEYS: {"ListID", "SendID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "list_subscriber":{
+                self.PRIMARY_KEYS: {"SubscriberKey", "ListID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "send":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "subscriber":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
```

Comment:
@harshpatel4crest Same question for 'ModifiedDate' here as well

Reply:
No, we are not getting it as it is not retrievable.

Reply:
If this is not retrievable we should discuss with product if it is expected. If this is a bug it should be written up in jira and marked in the test. But if this is a limitation in their api then we should probably make this a full table stream.

Reply:
@kspeer825 by taking reference of the [documentation](https://docs.google.com/document/d/1P4WV8uKMnPL9WHXlE49wAhzBSQxHDZjwHFfBcIZ7h5c/edit)and as per the discussion with @cosimon we have to keep the code as it is as the list_subscribers is an INCREMENTAL stream and the subscribers stream is a child stream of list_subscribers.
These streams work differently. The list_subscribers stream returns a list of subscriber_ids based on start date. These subscriber_ids are then passed to subscribers stream for getting the details of the subscriber. Hence, based on different start date subscriber_ids would vary. Hence, we will not be able to mention subscribers stream as a completely FULL_TABLE stream.

Reply:
I see, we do not have a good pattern for testing child streams yet. I think your approach makes sense, but could you add a comment that explains the relationship of `subscriptions` to `list_subscriptions` so that the expectations are clear.

Reply:
Added comment.